tgui/packages/tgui/interfaces/PersonalCooking.tsx

```diff
@@ -0,0 +1,481 @@
+                {item.name}
+              </Box>
+              <Box style={{ 'text-transform': 'capitalize' }} color={'gray'}>
+                {Array.from(
+                  item.reqs.map((i) =>
+                    i.is_reagent
+                      ? i.name + '\xa0' + i.amount + 'u'
+                      : i.amount > 1
+                        ? i.name + '\xa0' + i.amount + 'x'
+                        : i.name
+                  )
+                ).join(', ')}
+              </Box>
+            </Stack.Item>
+            <Stack.Item>
+              <Button
+                my={0.3}
+                lineHeight={2.5}
+                align="center"
+                content="Make"
+                disabled={!props.craftable || props.busy}
+                icon={props.busy ? 'circle-notch' : 'utensils'}
+                iconSpin={props.busy ? 1 : 0}
+                onClick={() =>
+                  act('make', {
+                    recipe: item.ref,
+                  })
+                }
+              />
+            </Stack.Item>
+          </Stack>
+        </Stack.Item>
+      </Stack>
+    </Section>
+  );
+};
+
+const RecipeContent = (props, context) => {
+  const { act } = useBackend<Data>(context);
+  const item = props.item;
+  return (
+    <Section>
+      <Stack>
+        <Stack.Item>
+          <Box width={'64px'} height={'64px'} mr={1}>
+            <Box
+              width={'32px'}
+              height={'32px'}
+              style={{
+                'transform': 'scale(2)',
+              }}
+              m={'16px'}
+              className={classes([
+                'food32x32',
+                item.result?.replace(/[/_]/gi, ''),
+              ])}
+            />
+          </Box>
+        </Stack.Item>
+        <Stack.Item grow>
+          <Stack>
+            <Stack.Item grow>
+              <Box bold mb={0.5} style={{ 'text-transform': 'capitalize' }}>
+                {item.name}
+              </Box>
+              <Box color={'gray'}>{item.desc}</Box>
+            </Stack.Item>
+            <Stack.Item>
+              <Button
+                width="104px"
+                lineHeight={2.5}
+                align="center"
+                content="Make"
+                disabled={!props.craftable || props.busy}
+                icon={props.busy ? 'circle-notch' : 'utensils'}
+                iconSpin={props.busy ? 1 : 0}
+                onClick={() =>
+                  act('make', {
+                    recipe: item.ref,
+                  })
+                }
+              />
+            </Stack.Item>
+          </Stack>
+          <hr style={{ 'border-color': '#111' }} />
+          <Stack>
+            <Stack.Item grow style={{ 'text-transform': 'capitalize' }}>
+              {item.reqs &&
+                item.reqs
+                  .sort((a, b) => (a.path > b.path ? 1 : -1))
+                  .map((item) => <IngredientContent key item={item} />)}
```

Comment:
1. Use sortBy
2. Always set the key to a unique value (ideally like item ref) and don't just leave it without a value - this can create many hard to find bugs with this list.